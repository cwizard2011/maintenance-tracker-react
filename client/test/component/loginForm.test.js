import React from 'react';
import configureMockStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import { shallow } from 'enzyme';
import DefaultLoginForm, { LoginForm } from '../../src/components/LoginForm';

const mockStore = configureMockStore([thunk]);

const props = {
  login: jest.fn(),
  userLogin: {
    loading: false,
    error: {}
  }
};
const initialState = {
  authReducer: {
    user: {
      email: 'juliet',
      exp: 1540907850,
      iat: 1540886250,
      id: '48a698a0-1641-5aca-bc1b-de9b1a482ee1',
      user_role: 'user',
      username: 'juliet'
    },
    isAuthenticated: false
  }
};
const store = mockStore(initialState);

describe('Login Form', () => {
  it('render login form correctly', () => {
    const wrapper = shallow(<LoginForm {...props} />);
    expect(wrapper).toMatchSnapshot();
  });
  it('should render connected component', () => {
    const wrapper = shallow(<DefaultLoginForm store={store} {...props} />);
    expect(wrapper.state()).toEqual({});
  });
  it('should call onSubmit', () => {
    const wrapper = shallow(<LoginForm {...props} />);
    const e = {
      preventDefault: jest.fn()
    };
    const instance = wrapper.instance();
    instance.onSubmit(e);
    expect(props.login).toHaveBeenCalled();
  });
  it('should call onchange event', () => {
    const event = {
      preventDefault: jest.fn(),
      target: {
        name: 'userInput',
        value: 'cwizard'
      }
    };
    const wrapper = shallow(<LoginForm {...props} />);
    const instance = wrapper.instance();
    instance.onChange(event);
    expect(wrapper.state().userInput).toEqual('cwizard');
  });
  it('should Render Loading when user submit login information', () => {
    const newProps = {
      ...props,
      userLogin: {
        loading: true,
      }
    };
    const wrapper = shallow(<LoginForm {...newProps} />);
    expect(wrapper.find('Loading')).toBeTruthy();
  });
  it('should Display login error code', () => {
    const newProps = {
      ...props,
      userLogin: {
        loading: false,
        error: {
          code: 401,
          message: 'Access Denied'
        },
      }
    };
    const wrapper = shallow(<LoginForm {...newProps} />);
    expect(wrapper.find('.red-text')).toBeTruthy();
  });
});
