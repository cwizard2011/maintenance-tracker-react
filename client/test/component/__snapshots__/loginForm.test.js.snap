// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login Form render login form correctly 1`] = `
<div>
  <div />
  <h2
    className="center-heading"
  >
    Login
  </h2>
  <form
    onSubmit={[Function]}
  >
    <label
      htmlFor="Username/Email"
    >
      Username/Email
      <input
        name="usernameInput"
        onChange={[Function]}
        placeholder="Username/Email"
        required={true}
        type="text"
        value=""
      />
    </label>
    <label
      htmlFor="Password"
    >
      Password
      <input
        name="password"
        onChange={[Function]}
        placeholder="Password"
        required={true}
        type="password"
        value=""
      />
    </label>
    <div
      className="red-text"
    />
    <input
      className="right-btn btn"
      disabled={true}
      type="submit"
      value="Login"
    />
  </form>
</div>
`;
